### PR TITLE
Update minimum flow setpoints for VAV terminals

### PR DESCRIPTION
**Pull request by lymereJ**
The proposed changes update the methods setting the minimum flow setpoint for VAV systems. They specifically focus on looking at the zone OA requirements to determine the setpoint.

### CODE REVIEW

File: lib/openstudio-standards/standards/Standards.AirLoopHVAC.rb [hunk 1]
```diff
@@ -1928,6 +1928,10 @@ def air_loop_hvac_apply_minimum_vav_damper_positions(air_loop_hvac, has_ddc = tr
           zone_oa = thermal_zone_outdoor_airflow_rate(zone)
           vav_terminal = equip.to_AirTerminalSingleDuctVAVReheat.get
           air_terminal_single_duct_vav_reheat_apply_minimum_damper_position(vav_terminal, zone_oa, has_ddc)
+        elsif equip.to_AirTerminalSingleDuctParallelPIUReheat.is_initialized
+          zone_oa = thermal_zone_outdoor_airflow_rate(zone)
+          fp_vav_terminal = equip.to_AirTerminalSingleDuctParallelPIUReheat.get
+          air_terminal_single_duct_parallel_piu_reheat_apply_minimum_primary_airflow_fraction(fp_vav_terminal, zone_oa)
```
Comment (lymereJ) [inline lib/openstudio-standards/standards/Standards.AirLoopHVAC.rb:1934]: The minimum primary air flow fraction was set during the FPB fan power calculation, the assignment was moved here for consistency.

File: lib/openstudio-standards/standards/Standards.AirTerminalSingleDuctParallelPIUReheat.rb [hunk 2]
```diff
@@ -58,4 +57,34 @@ def air_terminal_single_duct_parallel_piu_reheat_apply_prm_baseline_fan_power(ai
...
+
+    # Minimum OA flow rate
+    # If specified, set the primary air flow fraction as
+    unless zone_min_oa.nil?
+      min_primary_airflow_frac = [min_primary_airflow_frac, zone_min_oa / air_terminal_single_duct_parallel_piu_reheat.autosizedMaximumPrimaryAirFlowRate.get].max
+      air_terminal_single_duct_parallel_piu_reheat.setMinimumPrimaryAirFlowFraction(min_primary_airflow_frac)
+    end
+
+    return true
+  end
```
Comment (lymereJ) [inline lib/openstudio-standards/standards/Standards.AirTerminalSingleDuctParallelPIUReheat.rb:89]: New methods to handle the minimum primary air flow fraction for FPB.

File: lib/openstudio-standards/standards/Standards.AirTerminalSingleDuctVAVReheat.rb [hunk 3]
```diff
@@ -20,11 +20,10 @@ def air_terminal_single_duct_vav_reheat_apply_minimum_damper_position(air_termin
...
 
     # Minimum OA flow rate
-    # If specified, will also add this limit
-    # and the larger of the two will be used
-    # for sizing.
+    # If specified, set the MDP as the larger of the two
     unless zone_min_oa.nil?
-      air_terminal_single_duct_vav_reheat.setFixedMinimumAirFlowRate(zone_min_oa)
+      min_oa_damp_position = [zone_min_oa / air_terminal_single_duct_vav_reheat.autosizedMaximumAirFlowRate.get, min_damper_position].max
+      air_terminal_single_duct_vav_reheat.setConstantMinimumAirFlowFraction(min_oa_damp_position)
```
Comment (lymereJ) [inline lib/openstudio-standards/standards/Standards.AirTerminalSingleDuctVAVReheat.rb:26]: Take OA into account to determine the VAV box's minimum damper position.

File: lib/openstudio-standards/standards/Standards.Model.rb [hunk 4]
```diff
@@ -348,6 +341,11 @@ def model_create_prm_any_baseline_building(user_model, building_type, climate_zo
         return false
       end
 
+      # Apply the minimum damper positions, assuming no DDC control of VAV terminals
+      model.getAirLoopHVACs.sort.each do |air_loop|
+        air_loop_hvac_apply_minimum_vav_damper_positions(air_loop, false)
+      end
+
```
Comment (lymereJ) [inline lib/openstudio-standards/standards/Standards.Model.rb:348]: Move the loop so the methods is called after the sizing run with the final baseline HVAC system is available so the maximum air flow for terminals can be retrieved.

File: test/90_1_prm/test_appendix_g_prm.rb [hunk 5]
```diff
@@ -1666,8 +1666,39 @@ def check_number_of_cooling_towers(prototypes_base)
...
+              assert(expected_prim_frac == actual_prim_frac , "Minimum primary air flow fraction for #{building_type} for #{template} in #{climate_zone} should be #{expected_prim_frac} but #{actual_prim_frac} is used in the model.")
+            end
+          end
+        end
+      end
+    end  
+  end
+
+  # Set ZoneMultiplier to passed value for all zones
+  #
```
Comment (lymereJ) [inline test/90_1_prm/test_appendix_g_prm.rb:1701]: New test to check the MDP/Primary air flow fraction of two of the prototypes baseline models: medium office (FPB) and primary school (standard VAV boxes).

File: test/90_1_prm/test_appendix_g_prm.rb [hunk 6]
```diff
@@ -1701,8 +1701,39 @@ def check_number_of_cooling_towers(prototypes_base)
...
+              fp_vav_terminal = equip.to_AirTerminalSingleDuctParallelPIUReheat.get
+              expected_prim_frac = [zone_oa / fp_vav_terminal.autosizedMaximumPrimaryAirFlowRate.get, 0.3].max.round(2)
+              actual_prim_frac = fp_vav_terminal.minimumPrimaryAirFlowFraction.get
+              assert(expected_prim_frac == actual_prim_frac , "Minimum primary air flow fraction for #{building_type} for #{template} in #{climate_zone} should be #{expected_prim_frac} but #{actual_prim_frac} is used in the model.")
+            end
+          end
+        end
+      end
+    end  
+  end
```
Comment (lymereJ) [inline test/90_1_prm/test_appendix_g_prm.rb:1733]: New test to check the MDP/Primary air flow fraction of two of the prototypes baseline models: medium office (FPB) and primary school (standard VAV boxes).

File: lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.AirTerminalSingleDuctVAVReheat.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    min_damper_position = air_terminal_single_duct_vav_reheat_minimum_damper_position(air_terminal_single_duct_vav_reheat, has_ddc)
+    air_terminal_single_duct_vav_reheat.setConstantMinimumAirFlowFraction(min_damper_position)
+    OpenStudio.logFree(OpenStudio::Debug, 'openstudio.standards.AirTerminalSingleDuctVAVReheat', "For #{air_terminal_single_duct_vav_reheat.name}: set minimum damper position to #{min_damper_position}.")
+
+    # Minimum OA flow rate
+    # If specified, set the MDP as the larger of the two
+    unless zone_min_oa.nil?
+      min_oa_damp_position = [zone_min_oa / air_terminal_single_duct_vav_reheat.autosizedMaximumAirFlowRate.get, min_damper_position].max
+      air_terminal_single_duct_vav_reheat.setConstantMinimumAirFlowFraction(min_oa_damp_position)
+    end
```
Comment (lymereJ) [inline lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.AirTerminalSingleDuctVAVReheat.rb:27]: Take OA into account to determine the VAV box's minimum damper position.

**Review by dmaddoxwhite — APPROVED** [review]
Looks good!

**Pull request merged**